coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt

```diff
@@ -0,0 +1,276 @@
+                        duration = (crossfadeDuration * 0.75f).roundToInt()
+                    }
+                    saturation using tween {
+                        duration = crossfadeDuration
+                    }
+                }
+            }
+        }
+
+        // A new GetRequest will be created on every composition, which allows the requestBuilder
+        // to work on every invocation. executeAsComposable() guards the actual execution so that
+        // the request is only run if the request changes.
+        val result = GetRequest.Builder(ContextAmbient.current)
+                .data(data)
+                .size(constraints.requestWidth.value, constraints.requestHeight.value)
+                .apply {
+                    if (requestBuilder != null) {
+                        requestBuilder(this)
+                    }
+                }
+                .build()
+                .executeAsComposable()
+
+        if (result != null && imgLoadState == ImageLoadState.Empty) {
+            imgLoadState = ImageLoadState.Loaded
+        }
+
+        Transition(definition = transitionDef, toState = imgLoadState) { transitionState ->
+            if (result?.drawable != null) {
+                val asset = result.drawable!!.toBitmap().asImageAsset()
+
+                // Create and update the ImageLoadingColorMatrix from the transition state
+                val matrix = remember(asset) { ImageLoadingColorMatrix() }
+                matrix.saturationFraction = transitionState[saturation]
+                matrix.alphaFraction = transitionState[alpha]
+                matrix.brightnessFraction = transitionState[brightness]
+
+                // Unfortunately ColorMatrixColorFilter is not mutable so we have to create a new
+                // instance every time
+                val cf = ColorMatrixColorFilter(matrix)
+                Image(
+                    painter = AndroidColorMatrixImagePainter(asset, cf),
+                    contentScale = contentScale,
+                    alignment = alignment
+                )
+            }
+        }
+    }
+}
+
+/**
+ * Creates a composable that will attempt to load the given [data] using [Coil], and then
+ * display the result in an [Image].
+ *
+ * @param data The data to load. See [GetRequestBuilder.data] for the types allowed.
+ * @param requestBuilder Lambda which allows customization of the [GetRequestBuilder]
+ * @param alignment Optional alignment parameter used to place the loaded [ImageAsset] in the
+ * given bounds defined by the width and height.
+ * @param contentScale Optional scale parameter used to determine the aspect ratio scaling to be
+ * used if the bounds are a different size from the intrinsic size of the loaded [ImageAsset].
+ * @param modifier Modifier used to adjust the layout algorithm or draw decoration content (ex.
+ * background)
+ */
+@Composable
+fun CoilImage(
+    data: Any,
+    requestBuilder: ((GetRequestBuilder) -> Unit)? = null,
```

Comment:
Instead of having a `requestBuilder` lambda param what do you think about having `CoilImage` variants? One that accepts just `data: Any` for simple use cases and one that accepts a full `request: GetRequest`? Internally, you could use `newBuilder()` to modify the request instance. [Something like this](https://gist.github.com/colinrtwhite/05205d6c7fc0782883f9854c1ca290c3).

Reply:
SGTM, adding now.